packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -54,77 +53,88 @@ class ColdRunner extends ResidentRunner {
     }
 
     final String modeName = getModeName(debuggingOptions.buildMode);
-    if (mainPath == null) {
-      assert(prebuiltMode);
-      printStatus('Launching ${package.displayName} on ${device.name} in $modeName mode...');
-    } else {
-      printStatus('Launching ${getDisplayPath(mainPath)} on ${device.name} in $modeName mode...');
-    }
 
-    package = getApplicationPackageForPlatform(device.targetPlatform, applicationBinary: applicationBinary);
+    for (RunningDevice device in runningDevices) {
+      device.package = getApplicationPackageForPlatform(
```

Comment:
please refactor the code in this loop into a function that is called for each device.